### PR TITLE
Added autoAdjustHeight to core api

### PR DESCRIPTION
**Pull request by jeremye416**
I need to adjust the height of the grid based on the number of rows in my datasource. For example, set the height of the grid to the actual number of rows if less than 10, otherwise only show 10 and have the vertical scrollbar.

This behavior is easily accomplished using the minRowsToShow property. However, the grid doesn't respond to changes to the property after it is initialized.

Inside the compile function, autoAdjustHeight is called. I have fixed this by adding the autoAdjustHeight to the public api. Then I can call autoAdjustHeight when my datasource changes and I've updated the minRowstoShow property.

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
The tests need to be fixed as well.

**Comment by simonv3** [comment]
Looks like what the tests are complaining about is that the call to autoAdjustHeight() in init() references the old function, which was removed here.

**Comment by JewelsJLF** [comment]
What's the status of this?

**Comment by AnirudhaGohokar** [comment]
I think its still not merged with angular-ui master branch..

**Comment by jeremye416** [comment]
It should have been fixed yesterday.

**Comment by RiverMise** [comment]
Is there a new release with the fix?

**Comment by jeremye416** [comment]
Is there anything else I need to do to get this accepted?

**Comment by simonv3** [comment]
@jeremye416 looks like the branch is currently out of date. (though I'm not a maintainer, so I can't guarantee that that will make it get accepted)

**Comment by swalters** [comment]
@jeremye416 can you fix the merge conflicts, rebase to one commit, and force push again?

**Comment by jeremye416** [comment]
The merge / rebase / commit should be ready to go.

**Comment by swalters** [comment]
@jeremye416 I would really, really like to merge this. But, this has a lot of changes not related to your PR.  I think you must have included some changes that belong in other PR's.

**Comment by jeremye416** [comment]
I can try to start over. My change is only a couple of lines. I'll just
create a new branch from master and fold my changes back in.

**Comment by swalters** [comment]
@jeremye416 I may take a crack at a PR for this today unless you plan on a new PR in the next couple of hours.

File: src/js/core/factories/Grid.js [hunk 1]
```diff
@@ -2528,7 +2528,60 @@ angular.module('ui.grid')
...
+          if (!allColumnsHaveFilteringTurnedOff) {
+              maxNumberOfFilters++;
+          }
+      }
+
+      var filterHeight = maxNumberOfFilters * headerHeight;
+
+      var newHeight = headerHeight + contentHeight + footerHeight + scrollbarHeight + filterHeight;
+
+      var $elm = $(grid.element);
```
Comment (swalters) [inline src/js/core/factories/Grid.js]: jquery dependency not allowed

**Comment by swalters** [comment]
completed via https://github.com/angular-ui/ui-grid/commit/86d552cf9c300bf7e8843422ba05645f0521896c

**Comment by AnirudhaGohokar** [comment]
Hi @swalters,

Thanks for the commit.. Is this feature available in the latest UI grid v.3.1.1-d74dd46 ?

**Comment by simonv3** [comment]
Thanks @swalters!

**Pull request closed without merging**